### PR TITLE
agent: Add logging to libvirt qemu hook and cleanup

### PR DESCRIPTION
**Pull request by rhtyd**
This allows logging to the default libvirt qemu hook and cleans up the script.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

Installed the hook script on centos6 and xenial/16.04 based kvm hosts, changes the logging scope to `DEBUG`, performed VM migration, which caused the following logs:

```
16:57:25,61 qemu-hook DEBUG Executing qemu-hook with args: ['/etc/libvirt/hooks/qemu', 'i-2-39-VM', 'stopped', 'end', '-']
16:57:25,275 qemu-hook DEBUG Executing qemu-hook with args: ['/etc/libvirt/hooks/qemu', 'i-2-39-VM', 'release', 'end', '-']
16:57:34,685 qemu-hook DEBUG Executing qemu-hook with args: ['/etc/libvirt/hooks/qemu', 'i-2-39-VM', 'migrate', 'begin', '-']
16:57:34,904 qemu-hook DEBUG Executing qemu-hook with args: ['/etc/libvirt/hooks/qemu', 'i-2-39-VM', 'prepare', 'begin', '-']
16:57:36,880 qemu-hook DEBUG Executing qemu-hook with args: ['/etc/libvirt/hooks/qemu', 'i-2-39-VM', 'start', 'begin', '-']
16:57:37,388 qemu-hook DEBUG Executing qemu-hook with args: ['/etc/libvirt/hooks/qemu', 'i-2-39-VM', 'started', 'begin', '-']
```

## Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [ ] I have added tests to cover my changes.
- [x] All relevant new and existing integration tests have passed.
- [x] A full integration testsuite with all test that can run on my environment has passed.

@blueorangutan package

### CODE REVIEW

File: agent/bindir/libvirtqemuhook.in [hunk 1]
```diff
@@ -6,59 +6,78 @@
...
-import sys
+
+import logging
 import re
+import sys
 from xml.dom.minidom import parse
 from cloudutils.configFileOps import configFileOps
 from cloudutils.networkConfig import networkConfig
+
+logging.basicConfig(filename='/var/log/libvirt/qemu-hook.log',
```
Comment (rhtyd) [inline agent/bindir/libvirtqemuhook.in]: FYI - the directory `/var/log/libvirt/` exists if libvirt is installed.

Comment (DaanHoogland) [reply agent/bindir/libvirtqemuhook.in]: so, you mean no safe guard needed?

File: agent/bindir/libvirtqemuhook.in [hunk 2]
```diff
@@ -6,59 +6,78 @@
...
+
 if __name__ == '__main__':
     if len(sys.argv) != 5:
         sys.exit(0)
 
-    if sys.argv[2] == "migrate" and sys.argv[3] == "begin":
-        handleMigrateBegin() 
+    # For docs refer https://libvirt.org/hooks.html#qemu
+    logger.debug("Executing qemu hook with args: %s" % sys.argv)
+    guest_name, action, status = sys.argv[1:4]
```
Comment (DaanHoogland) [inline agent/bindir/libvirtqemuhook.in]: nice cleanup/readability improvement

Comment (DaanHoogland) [reply agent/bindir/libvirtqemuhook.in]: guest_name seems to drown, though.
action, status = sys.argv[2:4]
would suffice.

File: agent/bindir/libvirtqemuhook.in (hunk 1 above)
Comment (rhtyd) [reply agent/bindir/libvirtqemuhook.in]: Yes.

File: agent/bindir/libvirtqemuhook.in (hunk 2 above)
Comment (rhtyd) [reply agent/bindir/libvirtqemuhook.in]: I'll fix this.

File: agent/bindir/libvirtqemuhook.in [hunk 3]
```diff
@@ -6,59 +6,78 @@
...
+
 if __name__ == '__main__':
     if len(sys.argv) != 5:
         sys.exit(0)
 
-    if sys.argv[2] == "migrate" and sys.argv[3] == "begin":
-        handleMigrateBegin() 
+    # For docs refer https://libvirt.org/hooks.html#qemu
+    logger.debug("Executing qemu hook with args: %s" % sys.argv)
+    action, status = sys.argv[2:4]
```
Comment (rafaelweingartner) [inline agent/bindir/libvirtqemuhook.in]: Just to make sure I understood this notation here.  I was reading in the Internet about the notation to deal with arrays, and it seems that it is not inclusive (with respect of the first index used to define the range). I mean, if I want everything after the first parameter, which is the script name I can do the following:
`user_args = sys.argv[1:]`

Therefore, the code that we have here would assign parameter 3 to variable `action`, and parameter 4 to variable `status`. Is this true? I was under the impression that the variable `action` should be parameter 2, and `status` was expected to be parameter 3.

Comment (rafaelweingartner) [reply agent/bindir/libvirtqemuhook.in]: Ah.. never mind. After reading some more time the same explanation I think I understood now.
It is not inclusive with the upper bound parameter. Moreover, the arrays are indexed from 0. That is why if we want to ignore the script name, we start getting parameters that are in the first position.

**Comment by rhtyd** [comment]
@blueorangutan test

**Pull request merged**